src/rpdk/core/contract/suite/contract_asserts.py

```diff
@@ -139,3 +128,34 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator_wrapper
+
+
+@decorate
+def response_contains_resource_model_equal_current_model(
+    resource_client, input_model, output_model
+):
+    # only comparing properties in input model to those in output model and
+    # ignoring extraneous properties that maybe present in output model.
+    assertion_error_message = (
+        "All properties specified in the request MUST "
+        "be present in the model returned, and they MUST"
+        " match exactly, with the exception of properties"
+        " defined as writeOnlyProperties in the resource schema"
+    )
+    output_model = output_model["resourceModel"]
+    try:
+        for key in input_model:
+            if key in resource_client.properties_without_insertion_order:
```

Comment:
won't this only check top level lists without insertion order? what if there's a model like 
```
{
    "a": {
        "anotherList": ["a", "b", "c"]
    }
}
```